tests/integration/disputes_test.go

```diff
@@ -0,0 +1,337 @@
+		WonkyWorkReports:    mapHashes(psi.Wonky),
+		OffendingValidators: keys,
+	}
+}
+
+func mapKey(kappa Kappa) crypto.ValidatorKey {
+	return crypto.ValidatorKey{
+		Bandersnatch: crypto.BandersnatchPublicKey(stringToHex(kappa.Bandersnatch)),
+		Ed25519:      ed25519.PublicKey(stringToHex(kappa.Ed25519)),
+		Bls:          crypto.BlsKey(stringToHex(kappa.BLS)),
+		Metadata:     crypto.MetadataKey(stringToHex(kappa.Metadata)),
+	}
+}
+
+func mapJudgments(judgements []Judgement) [common.ValidatorsSuperMajority]block.Judgement {
+	var mappedJudgements [common.ValidatorsSuperMajority]block.Judgement
+	for i, judgement := range judgements {
+		mappedJudgements[i] = block.Judgement{
+			IsValid:        judgement.IsValid,
+			ValidatorIndex: uint16(judgement.ValidatorIndex),
+			Signature:      crypto.Ed25519Signature(stringToHex(judgement.Signature)),
+		}
+	}
+	return mappedJudgements
+}
+
+func mapVerdicts(verdicts []Verdict) []block.Verdict {
+	mappedVerdicts := make([]block.Verdict, len(verdicts))
+	for i, verdict := range verdicts {
+		mappedVerdicts[i] = block.Verdict{
+			ReportHash: crypto.Hash(stringToHex(verdict.ReportHash)),
+			EpochIndex: uint32(verdict.EpochIndex),
+			Judgements: mapJudgments(verdict.Judgements),
+		}
+	}
+	return mappedVerdicts
+}
+
+func mapCulprits(culprits []Culprit) []block.Culprit {
+	mappedCulprits := make([]block.Culprit, len(culprits))
+	for i, culprit := range culprits {
+		mappedCulprits[i] = block.Culprit{
+			ReportHash:                crypto.Hash(stringToHex(culprit.ReportHash)),
+			ValidatorEd25519PublicKey: ed25519.PublicKey(stringToHex(culprit.ValidatorEd25519PublicKey)),
+			Signature:                 crypto.Ed25519Signature(stringToHex(culprit.Signature)),
+		}
+	}
+	return mappedCulprits
+}
+
+func mapFaults(faults []Fault) []block.Fault {
+	mappedFaults := make([]block.Fault, len(faults))
+	for i, fault := range faults {
+		mappedFaults[i] = block.Fault{
+			ReportHash:                crypto.Hash(stringToHex(fault.ReportHash)),
+			IsValid:                   fault.IsValid,
+			ValidatorEd25519PublicKey: ed25519.PublicKey(stringToHex(fault.ValidatorEd25519PublicKey)),
+			Signature:                 crypto.Ed25519Signature(stringToHex(fault.Signature)),
+		}
+	}
+	return mappedFaults
+}
+
+func mapDisputes(disputes Disputes) block.DisputeExtrinsic {
+	return block.DisputeExtrinsic{
+		Verdicts: mapVerdicts(disputes.Verdicts),
+		Culprits: mapCulprits(disputes.Culprits),
+		Faults:   mapFaults(disputes.Faults),
+	}
+}
+
+func TestDisputes(t *testing.T) {
+	files, err := os.ReadDir("vectors/disputes/tiny")
+	require.NoError(t, err, "failed to read tiny directory")
+
+	for _, file := range files {
+		if !strings.HasSuffix(file.Name(), ".json") {
+			continue
+		}
+
+		t.Run(file.Name(), func(t *testing.T) {
+			filePath := fmt.Sprintf("vectors/disputes/tiny/%s", file.Name())
+			data, err := ReadJSONFile(filePath)
+			require.NoError(t, err, "failed to read JSON file: %s", filePath)
```

Comment:
Only nitpick with this is that it would be nice to also see the test description somewhere.

Reply:
Agree, but didn't want to hardcode the files. I plan on open another one for the latest version of the GP, might add it as just a comment.